src/mixins/form-size.js

```diff
@@ -1,9 +1,11 @@
+import { getComponentConfig } from '../utils/config'
+
 // @vue/component
 export default {
   props: {
     size: {
       type: String,
-      default: null
+      default: () => getComponentConfig('BFormControl', 'size')
```

Comment:
Wonder if `this` is available when the default is evaluation....

Could use `default: () => getComponentConfig(this.$options.name, 'size')` if it is available.

Reply:
But then we need to add the size to the config for each component.

Reply:
Or we use a lowercase name for this key (similar to `breakpoints`), as `BFormControl` looks like a component that doesn't exist.

Reply:
@tmorehouse This would work:

```js
default() {
  return getComponentConfig(this.$options.name, 'size')
}
```

Reply:
Or maybe this:

```js
default() {
  return getComponentConfig(this.$options.name, 'size') || getComponentConfig('formControls`, 'size')
}
```

so it has a fallback

Reply:
The only time this wouldn't work:

```js
default() {
  return getComponentConfig(this.$options.name, 'size')
}
```

Would be if a consumer `extends` a component and gives it a new name...

Reply:
I thin `formControls` is good for now.

Reply:
yeah...